client/state/data-getters/index.js

```diff
@@ -100,3 +100,31 @@ export const requestSiteAlerts = siteId => {
 		}
 	);
 };
+
+export const requestGutenbergDraftPost = siteId =>
+	requestHttpData(
+		`gutenberg-draft-post-${ siteId }`,
```

Comment:
I don't believe this key is sufficient. For example, what happens when we load the new post page multiple times? We probably need to assign a temporary draft id, until the response comes back.

Reply:
If we look at media for example, I believe the transient object we create is `uniqueId( 'media-' ),`

We only need to make this unique to the client, so we can do something similar here with:
```
uniqueId( `gutenberg-draft-post-${ siteId }-`
```

Reply:
Careful! It looks like we're also creating a new post from a GET request.

Reply:
Fixed the `GET` verb on d77f521a461c0b9ba05550712e20f8356b69e719 and updated D18064-code also. Please apply the patch again before testing.

About the `requestId`, I don't think it'll be a problem to use that key. I haven't found any problems (blocked requests) when reloading the page or loading multiple pages. But using an `uniqueId` is always a good idea.

I've implemented two versions, one using the route's controller (d2d95b6fcdf978ecefc68db4fe792e299a10daa3):

```js
export const post = ( context, next ) => {
	const postId = getPostID( context );
	const uniqueDraftId = uniqueId( 'gutenberg-draft-' );

	//reserved space for adding the post to context see post-editor/controller.js
	context.primary = <GutenbergEditor { ...{ postId, uniqueDraftId } } />;
	next();
};
```

 and another using a HOC (fd77025afdb0c1d372155fb900c0fc036da7e10b, e8899f785f64af9402488b287312536dc800e608 and 714beb69a514e8ec2d9a71a038a4198d9925155c):

```jsx
export const withUniqueDraftId = WrappedComponent => {
	return class extends Component {
		constructor( props ) {
			super( props );

			this.state = {
				uniqueDraftId: uniqueId( 'gutenberg-draft-' ),
			};
		}

		render() {
			const { uniqueDraftId } = this.state;

			return <WrappedComponent { ...{ ...this.props, uniqueDraftId } } />;
		}
	};
};
```

```js
export default withUniqueDraftId( connect( mapStateToProps )( GutenbergEditor ) );
```

You can verify that the `requestId` is passed down to the `HTTP_DATA_REQUEST` action on the redux inspector:

![screen shot 2018-09-12 at 00 17 06](https://user-images.githubusercontent.com/233601/45400166-76510f80-b621-11e8-9e80-d57bf7913a06.png)

which version do we prefer? I'm leaning towards using the controller just because it's far less code, but composing things is nice too.